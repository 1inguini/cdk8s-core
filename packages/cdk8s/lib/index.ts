export * from './service';
export * from './deployment';
export * from './api-object';
export * from './chart';
export * from './horizontal-pod-autoscaler';
export * from './ingress';