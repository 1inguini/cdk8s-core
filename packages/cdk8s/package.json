{
  "name": "@awslabs/cdk8s",
  "version": "0.0.0",
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  },
  "description": "Cloud Development Kit for Kubernetes",
  "scripts": {
    "build": "./generate.sh && tsc",
    "watch": "tsc -w",
    "package": "./package.sh",
    "test": "jest"
  },
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "jsii": {
    "outdir": "dist",
    "targets": {
      "java": {
        "package": "com.github.eladb.cdk8s",
        "maven": {
          "groupId": "com.github.eladb",
          "artifactId": "cdk8s"
        }
      },
      "python": {
        "distName": "cdk8s",
        "module": "cdk8s"
      }
    }
  },
  "author": {
    "email": "elad.benisrael@gmail.com",
    "name": "Elad Ben-Israel",
    "url": "https://github.com/eladb"
  },
  "repository": {
    "type": "git",
    "url": "git://github.com/awslabs/cdk8s.git"
  },
  "license": "Apache-2.0",
  "devDependencies": {
    "@types/jest": "^25.1.2",
    "@types/node": "13.7.0",
    "@types/yaml": "^1.2.0",
    "jest": "^25.1.0",
    "jsii": "^0.22.0",
    "jsii-pacmak": "^0.22.0",
    "json-schema-to-typescript": "^8.0.1",
    "typescript": "^3.7.5"
  },
  "jest": {
    "moduleFileExtensions": [
      "js"
    ]
  },
  "bundledDependencies": [
    "yaml"
  ],
  "dependencies": {
    "yaml": "^1.7.2"
  },
  "peerDependencies": {
    "@aws-cdk/core": "^1.23.0",
    "@aws-cdk/cx-api": "^1.23.0"
  }
}
