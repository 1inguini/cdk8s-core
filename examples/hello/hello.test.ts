import { App } from '@aws-cdk/core';
import { HelloKube } from './hello';
import { Testing } from '@awslabs/cdk8s';

test('snapshot', () => {
  const app = new App();
  const chart = new HelloKube(app, 'hello');

  expect(Testing.synth(chart)).toMatchSnapshot();
});