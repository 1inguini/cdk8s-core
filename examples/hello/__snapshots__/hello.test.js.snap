// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "helloservice55622b70",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
        },
      ],
      "selector": Object {
        "app": "hello-k8s",
      },
      "type": "LoadBalancer",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "hellodeployment8f520003",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "hello-k8s",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "hello-k8s",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "paulbouwer/hello-kubernetes:1.5",
              "name": "hello-kubernetes",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
            },
          ],
        },
      },
    },
  },
]
`;
